### ISSUE TITLE
Possible change to learn.recipe

### ISSUE THREAD

**Issue by ctlente** [topepo/recipes#47]
Hey there!

I've been using your package for a over week now and I love it! I found it so useful that I started creating my own `step_shatter` function.

While adding functionalities to my custom step, I came across a problem: whenever I ran `step_shatter` as the 2nd step in a recipe, things would break apart. After a little debugging, I discovered that `info` didn't contain all the information I needed.

To fix the problem, all I had to do was change this line in `recipe.R`:

`x$steps[[i]] <- learn(x$steps[[i]], training = training, info = x$term_info)`

To this:

`x$steps[[i]] <- learn(x$steps[[i]], training = training, info = x$var_info)`

Would this break any functionality? If not, can I create a PR?

**Comment by topepo** [comment]
The difference between `term_info` and `var_info` is that the latter only contains information on the original variables. For example:
```r
> data(biomass)
> 
> # split data
> biomass_tr <- biomass[biomass$dataset == "Training",]
> biomass_te <- biomass[biomass$dataset == "Testing",]
> 
> rec <- recipe(HHV ~ carbon + hydrogen + oxygen + nitrogen + sulfur,
+               data = biomass_tr) %>%
+   step_center(all_predictors())
> 
> rec <- learn(rec, biomass_tr)
step 1 center training 
> 
> summary(rec, original = TRUE)
# A tibble: 6 × 4
  variable    type      role   source
     <chr>   <chr>     <chr>    <chr>
1   carbon numeric predictor original
2 hydrogen numeric predictor original
3   oxygen numeric predictor original
4 nitrogen numeric predictor original
5   sulfur numeric predictor original
6      HHV numeric   outcome original
> summary(rec, original = FALSE)
# A tibble: 6 × 4
  variable    type      role   source
     <chr>   <chr>     <chr>    <chr>
1   carbon numeric predictor original
2 hydrogen numeric predictor original
3   oxygen numeric predictor original
4 nitrogen numeric predictor original
5   sulfur numeric predictor original
6      HHV numeric   outcome original
> 
> rec <- rec %>% step_pca(all_numeric(), num = 3)
> rec <- learn(rec, biomass_tr)
step 1 center [pre-trained]
step 2 pca training 
> 
> summary(rec, original = TRUE)
# A tibble: 6 × 4
  variable    type      role   source
     <chr>   <chr>     <chr>    <chr>
1   carbon numeric predictor original
2 hydrogen numeric predictor original
3   oxygen numeric predictor original
4 nitrogen numeric predictor original
5   sulfur numeric predictor original
6      HHV numeric   outcome original
> summary(rec, original = FALSE)
# A tibble: 3 × 4
  variable    type      role  source
     <chr>   <chr>     <chr>   <chr>
1      PC1 numeric predictor derived
2      PC2 numeric predictor derived
3      PC3 numeric predictor derived
```
This comes from:
```r
> recipes:::summary.recipe
function (object, original = FALSE, ...) 
{
    if (original) 
        object$var_info
    else object$term_info
}
```

The `learn` function uses `term_info` as it goes to make sure that subsequent steps use what is currently in the data. 

Can you give an example where your step breaks down?

**Comment by ctlente** [comment]
Thank you for the prompt response.

I found out why my code wasn't working, it had nothing to do with `term_info` and `var_info`. I was using a logical variable as outcome and it wasn't passed through to `step_shatter` via `info`; once I converted the outcome variable to factor, everything started working as intended.

I think this is the expected behavior, so I'm closing the issue. Thank you very much.
